straxen/plugins/acqmon_processing.py

```diff
@@ -130,4 +133,82 @@ def merge_vetos(channels, gap, dtype, t):
 @numba.njit
 def channel_select(rr, ch_stop, ch_start):
     """Return data in between stop and start channels of the acquisition monitor (AM)"""
-    return rr[((rr['channel'] == ch_stop) | (rr['channel'] == ch_start))] 
+    return rr[((rr['channel'] == ch_stop) | (rr['channel'] == ch_start))]
+
+
+
+class VetoProximity(strax.Plugin):
+    """ 
+    Find the closest next/previous veto start and end to each event center.
+    
+    previous_x: Time in ns between the time center of an event and the previous x
+    next_x: Time in ns between the time center of an event and the next x
+    This also considers any x inside the event. x could be either:
+        - busy_x: busy on/off signal
+        - he_x:   high energy channels busy on/off signal
+        - hev_x:  high energy veto on/off signal
+    """
+        
+    __version__ = '0.0.1'
+    depends_on = ('veto_intervals','events') 
+    provides  = ('veto_proximity')
+    data_kind = ('veto_proximity')
+    
+    veto_names = ['busy', 'he', 'hev']
+    
+    def infer_dtype(self):
+        dtype = []
+        for n in self.veto_names:
+            dtype.append((('Time to previous %s veto start [ns]' %n, 'previous_%s_on' %n), np.int64))
+            dtype.append((('Time to previous %s veto end [ns]' %n, 'previous_%s_off' %n), np.int64))
+            dtype.append((('Time to next %s veto start [ns]' %n, 'next_%s_on' %n), np.int64))
+            dtype.append((('Time to next %s veto end [ns]' %n, 'next_%s_off' %n), np.int64))
+        
+        dtype += strax.time_fields
+
+        return dtype
+    
+    def setup(self):
+        self.states = ['on', 'off']
+        
+    def compute(self, events, veto_intervals):
+        res = defaultdict(list)
+        
+        t_event_centers = (events['time'] + events['endtime'])//2
+
+        for n, name in enumerate(self.veto_names):
+            veto_selection = veto_intervals[veto_intervals['veto_type'] == ('%s_veto' %name)]
+            
+            # For each state find the next and previous veto
+            for state in self.states:
+                prev = 'previous_%s_%s' %(name, state)
+                nxt = 'next_%s_%s' %(name, state)
+                
+                if state == 'on':
+                    veto_start_time_selection = veto_selection['time']
+                else:
+                    veto_start_time_selection = veto_selection['endtime']
+                    
+                for t, time in enumerate(t_event_centers):
+                    inx = np.searchsorted(veto_start_time_selection, time, side = 'left')
+                    
+                    # Time to previous veto on/off
+                    # Just using maxsize as a huge value that will not fit in any potential VetoCut range
+                    if inx == 0:
+                        res[prev].append(sys.maxsize)
```

Comment:
Perhaps this is slightly overkill as it might depend on your machine and not fit in the np.int64. Perhaps define some hard coded improbable value in this python file such that you can do ` veto_proximity['next_busy_on']==straxen.acqmon_processing.T_NO_VETO_FOUND` .

Reply:
I guess I can think of something else. I was using np.inf, but strax.dict_to_rec didn't like it.